examples/workbooks/LeadsExample.ts

```diff
@@ -0,0 +1,324 @@
+ *
+ * @example
+ * pipe(value, toLowerCase);
+ */
+const toLowerCase = (value: string): string => value.toLowerCase();
+
+/**
+ * Converts `String.prototype.trim()` to a normal fn so it can be used with `pipe`.
+ *
+ * @param {string} value - value to apply operation.
+ *
+ * @example
+ * pipe(value, trim);
+ */
+const trim = (value: string): string => value.trim();
+
+/**
+ * Allows us to combine multiple validations in a quick and easy way.
+ *
+ * @example
+ * runValidations(fn1, fn2, fn3, ...);
+ */
+const runValidations = (...fns: Array<any>): Array<FF.Message> => {
+    return fns.reduce((acc, fn) => [...acc, fn()], []).filter(isNotNil);
+};
+
+/**
+ * Allows us to sequence multiple RecordHooks _synchronously_ on a `FlatfileRecord`.
+ *
+ * @example
+ * runRecordHooks(fn1, fn2, fn3, ...)(record)
+ */
+export const runRecordHooks =
+    (...fns: Array<(x: FlatfileRecord) => void>) =>
+        (x: FlatfileRecord) =>
+            fns.map((f) => f(x));
+
+
+/*
+ * Main
+ */
+
+const LeadsSheet = new FF.Sheet(
+    "Leads",
+    {
+        first_name: FF.TextField({
+            label: "First Name",
+            description: "Lead's first name.",
+        }),
+        last_name: FF.TextField({
+            label: "Last Name",
+            description: "Lead's last name.",
+            required: true,
+        }),
+        email: FF.TextField({
+            label: "Email Address",
+            description: "Lead's email.",
+            unique: true,
+            compute: (value) => {
+                return pipe(value, trim, toLowerCase);
+            },
+            validate: (value) => {
+                const ensureValidEmail = validateEmail(value);
+
+                return runValidations(ensureValidEmail);
+            },
+        }),
+        phone: FF.TextField({
+            label: "Phone Number",
+            description: "Lead's phone.",
+        }),
+        date: FF.DateField({
+            label: "Date",
+            description: "Date goes here.",
+        }),
+        country: FF.TextField({
+            label: "Country",
+            description: "Country goes here",
+            cast: FF.CountryCast("iso-2"),
+        }),
+        postal_code: FF.TextField({
+            label: "Postal Code",
+            description: "Postal code goes here",
+        }),
+        opt_in: FF.BooleanField({
+            label: "Opt In",
+            description: "Opt in goes here",
+        }),
+        deal_status: FF.OptionField({
+            label: "Deal Status",
+            description: "Deal status goes here",
+            options: {
+                prospecting: "Prospecting",
+                discovery: "Discovery",
+                proposal: "Proposal",
+                negotiation: "Negotiation",
+                closed_won: "Closed Won",
+                closed_lost: "Closed Lost",
+            },
+        }),
+    },
+    {
+        allowCustomFields: false,
+        readOnly: true,
```

Comment:
I'd remove "readOnly" from this. We learned it doesn't actually do anything and could be confusing.